core/src/main/java/it/pagopa/selfcare/onboarding/core/InstitutionServiceImpl.java

```diff
@@ -86,7 +92,7 @@ public void onboardingProduct(OnboardingData onboardingData) {
         Assert.notNull(onboardingData, REQUIRED_ONBOARDING_DATA_MESSAGE);
         Assert.notNull(onboardingData.getBilling(), REQUIRED_INSTITUTION_BILLING_DATA_MESSAGE);
         Assert.notNull(onboardingData.getInstitutionType(), REQUIRED_INSTITUTION_TYPE_MESSAGE);
-
+        checkProductAvaibility(onboardingData.getProductId(), onboardingData.getInstitutionType(), onboardingData.getInstitutionExternalId());
```

Comment:
you could refactor the method adding check if the product is ACTIVE.....could exist only one method es. "checkProductStatus" that check if a product is ACTIVE and if the institution is a PT has new status mentioned below as ACTIVE